Mac compile error: MSG_NOSIGNAL

Not sure whether cnping or cnping_mac is the correct target, but both don't compile on macOS (10.13):

```
$ make cnping
cc  -g -Os -I/opt/X11/include -Wall   -c -o httping.o httping.c
httping.c:96:36: error: use of undeclared identifier 'MSG_NOSIGNAL'
                int rs = send( httpsock, buf, n, MSG_NOSIGNAL );
                                                 ^
1 error generated.
make: *** [<builtin>: httping.o] Error 1
```


```
$ make cnping_mac
gcc -o cnping cnping.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c -x objective-c -framework Cocoa -framework QuartzCore -lm -lpthread
cnping.c:344:27: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                        sprintf( stbuf, "%lu", samps );
                                         ~~~   ^~~~~
                                         %llu
/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
cnping.c:358:55: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                        pinghost, globmaxtime*1000.0, globinterval*1000.0, globallost, globalrx, globallost*100.0/(globalrx+globallost) );
                                                                           ^~~~~~~~~~
/usr/include/secure/_stdio.h:57:62: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                                                             ^~~~~~~~~~~
cnping.c:358:67: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                        pinghost, globmaxtime*1000.0, globinterval*1000.0, globallost, globalrx, globallost*100.0/(globalrx+globallost) );
                                                                                       ^~~~~~~~
/usr/include/secure/_stdio.h:57:62: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                                                             ^~~~~~~~~~~
cnping.c:478:3: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                globallost, globalrx, globallost*100.0f/(globalrx+globallost), loss );
                ^~~~~~~~~~
/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
cnping.c:478:15: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
                globallost, globalrx, globallost*100.0f/(globalrx+globallost), loss );
                            ^~~~~~~~
/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
5 warnings generated.
os_generic.c:281:2: warning: 'sem_init' is deprecated [-Wdeprecated-declarations]
        sem_init( sem, 0, 0 );
        ^
/usr/include/sys/semaphore.h:55:42: note: 'sem_init' has been explicitly marked deprecated here
int sem_init(sem_t *, int, unsigned int) __deprecated;
                                         ^
/usr/include/sys/cdefs.h:176:37: note: expanded from macro '__deprecated'
#define __deprecated    __attribute__((deprecated))
                                       ^
os_generic.c:288:2: warning: 'sem_getvalue' is deprecated [-Wdeprecated-declarations]
        sem_getvalue( os, &valp );
        ^
/usr/include/sys/semaphore.h:54:56: note: 'sem_getvalue' has been explicitly marked deprecated here
int sem_getvalue(sem_t * __restrict, int * __restrict) __deprecated;
                                                       ^
/usr/include/sys/cdefs.h:176:37: note: expanded from macro '__deprecated'
#define __deprecated    __attribute__((deprecated))
                                       ^
os_generic.c:305:2: warning: 'sem_destroy' is deprecated [-Wdeprecated-declarations]
        sem_destroy( os );
        ^
/usr/include/sys/semaphore.h:53:26: note: 'sem_destroy' has been explicitly marked deprecated here
int sem_destroy(sem_t *) __deprecated;
                         ^
/usr/include/sys/cdefs.h:176:37: note: expanded from macro '__deprecated'
#define __deprecated    __attribute__((deprecated))
                                       ^
3 warnings generated.
ping.c:58:4: warning: declaration does not declare anything [-Wmissing-declarations]
                        uint16_t        __unused;
                        ^~~~~~~~
1 warning generated.
Undefined symbols for architecture x86_64:
  "_StartHTTPing", referenced from:
      _main in cnping-a4c237.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:24: cnping_mac] Error 1
```

The second error can be reduced to the first with:
```diff
diff --git a/Makefile b/Makefile
index 63f799e..b0c9fcc 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ cnping : cnping.o CNFGFunctions.o CNFGXDriver.o os_generic.o ping.o httping.o
 cnping-mousey : cnping-mousey.o CNFGFunctions.o CNFGXDriver.o os_generic.o ping.o
        gcc $(CFLAGS) -o $@ $^ -lX11 -lm -lpthread $(LDFLAGS)

-cnping_mac : cnping.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c
+cnping_mac : cnping.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c httping.c
        gcc -o cnping $^ -x objective-c -framework Cocoa -framework QuartzCore -lm -lpthread

 cnping-mousey_mac : cnping-mousey.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c
```

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ cnping.exe : cnping.c CNFGFunctions.c CNFGWinDriver.c os_generic.c ping.c httpin
 cnping : cnping.o CNFGFunctions.o CNFGXDriver.o os_generic.o ping.o httping.o
 	gcc $(CFLAGS) -o $@ $^ -lX11 -lm -lpthread $(LDFLAGS) 
 
-cnping_mac : cnping.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c
+cnping_mac : cnping.c CNFGFunctions.c CNFGCocoaCGDriver.m os_generic.c ping.c httping.o
 	gcc -o cnping $^ -x objective-c -framework Cocoa -framework QuartzCore -lm -lpthread
 
 searchnet : os_generic.o ping.o searchnet.o

--- a/httping.c
+++ b/httping.c
@@ -6,12 +6,12 @@
 #include <unistd.h>
 #include <sys/types.h>
 
+#ifndef MSG_NOSIGNAL
+	#define MSG_NOSIGNAL 0
+#endif
+
 #ifdef WIN32
 	#include <winsock2.h>
-	#ifndef MSG_NOSIGNAL 
-	#define MSG_NOSIGNAL 0
-	#endif
-
 #else
 	#include <sys/socket.h>
 	#include <netinet/in.h>
@@ -85,6 +85,11 @@ reconnect:
 		ERRMB( "%s: ERROR connecting\n", hostname );
 		goto fail;
 	}
+
+#ifdef __APPLE__
+	int opt = 1;
+	setsockopt(httpsock, SOL_SOCKET, SO_NOSIGPIPE, &opt, sizeof(opt));
+#endif
 
 	errbuffer[0] = 0;
 

